app/src/backlight.c

```diff
@@ -42,56 +39,41 @@ static struct backlight_state state = {.brightness = CONFIG_ZMK_BACKLIGHT_BRT_ST
 
 static int zmk_backlight_update() {
     uint8_t brt = state.on ? state.brightness : 0;
-    for (int i = 0; i < BACKLIGHT_NUM_LEDS; i++) {
-        int rc = led_set_brightness(backlight_dev, i, brt);
-        if (rc != 0) {
-            return rc;
-        }
-    }
-    return 0;
+    return led_set_brightness(backlight_dev, 0, brt);
 }
 
 #if IS_ENABLED(CONFIG_SETTINGS)
 static int backlight_settings_set(const char *name, size_t len, settings_read_cb read_cb,
                                   void *cb_arg) {
     const char *next;
+    int rc;
 
     if (settings_name_steq(name, "state", &next) && !next) {
         if (len != sizeof(state)) {
             return -EINVAL;
         }
 
-        int rc = read_cb(cb_arg, &state, sizeof(state));
-        if (rc < 0) {
-            return rc;
+        rc = read_cb(cb_arg, &state, sizeof(state));
+        if (rc >= 0) {
+            return 0;
         }
 
-        return zmk_backlight_update();
+        return rc;
     }
 
     return -ENOENT;
 }
 
-static struct settings_handler backlight_conf = {.name = "backlight",
-                                                 .h_set = backlight_settings_set};
+struct settings_handler backlight_conf = {.name = "backlight", .h_set = backlight_settings_set};
```

Comment:
This variable should be declared as static.

Reply:
Will fix